core/include/byteorder.h

```diff
@@ -460,12 +485,28 @@ static inline uint16_t byteorder_bebuftohs(const uint8_t *buf)
     return (uint16_t)((buf[0] << 8) | (buf[1] << 0));
 }
 
+static inline uint32_t byteorder_bebuftohl(const uint8_t *buf)
+{
+    return (uint32_t)((buf[0] << 24)
```

Comment:
The cast needs to go before the shift, at least for the first one, otherwise you are shifting out of scope:

```suggestion
    return (((uint32_t)buf[0] << 24)
```

Reply:
Ping @jia200x ?

Reply:
you are right! I will fix it

Reply:
wait... does that mean `byteorder_bebuftohs` has the same issue?

Reply:
It's only an issue if the shift is >= the width of the type.
`buf[0]`will be promoted to `int`which is only 16 bit on some platforms.

So shifting by 8 is fine, shifting by 24 will be an issue.